### PR TITLE
[offer processor] Default to using the network PK if the shop's config does not have a PK

### PR DESCRIPTION
**Pull request by franckc**
- Default to using Network PK
 - Capture exceptions in Sentry

### CODE REVIEW

**Comment by nick** [comment]
Hrm not sure I see the relevant change in here... looks like we just added sentry?

File: backend/queues/makeOfferProcessor.js [hunk 1]
```diff
@@ -23,46 +27,55 @@ function attachToQueue() {
...
+    queueLog(10, 'Creating offer')
+    const lid = ListingID.fromFQLID(shop.listingId)
+    const offer = createOfferJson(lid, encryptedData, paymentCode)
+    const ires = await postOfferIPFS(network, offer)
+
+    queueLog(20, 'Submitting Offer')
+    const web3 = new Web3(network.provider)
+
+    // Use the Shop PK if there is one, otherwise fall back to Network PK.
+    const backendPk = shopConfig.web3Pk || networkConfig.web3Pk
```
Comment (mikeshultz) [inline backend/queues/makeOfferProcessor.js:54]: Maybe we should make this more defensive?  What if a deployment doesn't have a privkey configured for the network yet?

Comment (franckc) [reply backend/queues/makeOfferProcessor.js:54]: good to be defensive. let me check and throw an error...

Comment (franckc) [reply backend/queues/makeOfferProcessor.js:59]: added

Comment (mikeshultz) [reply backend/queues/makeOfferProcessor.js:59]: That doesn't address my concern above?  What happens if neither `shopConfig.web3Pk` nor `networkConfig.web3Pk` are anything on someone's custom deployment?

**Pull request merged**